kak-lsp.toml

```diff
@@ -6,163 +6,184 @@
 # dialyzerEnable = true
 
-[language.elm]
+[language_server.elm-language-server]
 filetypes = ["elm"]
 roots = ["elm.json"]
 command = "elm-language-server"
 args = ["--stdio"]
 settings_section = "elmLS"
-[language.elm.settings.elmLS]
+[language_server.elm-language-server.settings.elmLS]
 # See https://github.com/elm-tooling/elm-language-server#server-settings
 runtime = "node"
 elmPath = "elm"
 elmFormatPath = "elm-format"
 elmTestPath = "elm-test"
 
-[language.elvish]
+[language_server.elvish]
 filetypes = ["elvish"]
 roots = [".git", ".hg"]
 command = "elvish"
 args = ["-lsp"]
 
-[language.erlang]
+[language_server.erlang-ls]
 filetypes = ["erlang"]
 # See https://github.com/erlang-ls/erlang_ls.git for more information and
 # how to configure. This default config should work in most cases though.
 roots = ["rebar.config", "erlang.mk", ".git", ".hg"]
 command = "erlang_ls"
 
-[language.go]
+[language_server.gopls]
 filetypes = ["go"]
 roots = ["Gopkg.toml", "go.mod", ".git", ".hg"]
 command = "gopls"
-settings_section = "gopls"
-[language.go.settings.gopls]
+[language_server.gopls.settings.gopls]
 # See https://github.com/golang/tools/blob/master/gopls/doc/settings.md
 # "build.buildFlags" = []
 
-[language.haskell]
+[language_server.haskell-language-server]
 filetypes = ["haskell"]
 roots = ["hie.yaml", "cabal.project", "Setup.hs", "stack.yaml", "*.cabal"]
 command = "haskell-language-server-wrapper"
 args = ["--lsp"]
 settings_section = "_"
-[language.haskell.settings._]
+[language_server.haskell-language-server.settings._]
 # See https://haskell-language-server.readthedocs.io/en/latest/configuration.html
 # haskell.formattingProvider = "ormolu"
 
-[language.html]
+[language_server.html-language-server]
 filetypes = ["html"]
 roots = ["package.json"]
 command = "vscode-html-languageserver"
 args = ["--stdio"]
 
-# # Commented out by default because you still need to set the paths in the JDT
-# # Language Server arguments below before this can become a valid configuration.
-# [language.java]
-# filetypes = ["java"]
-# roots = [".git", "mvnw", "gradlew"]
-# command = "java"
-# args = [
-#     "-Declipse.application=org.eclipse.jdt.ls.core.id1",
-#     "-Dosgi.bundles.defaultStartLevel=4",
-#     "-Declipse.product=org.eclipse.jdt.ls.core.product",
-#     "-Dlog.level=ALL",
-#     "-Dfile.encoding=utf-8",
-#     "--add-modules=ALL-SYSTEM",
-#     "--add-opens",
-#     "java.base/java.util=ALL-UNNAMED",
-#     "--add-opens",
-#     "java.base/java.lang=ALL-UNNAMED",
-#     "-noverify",
-#     "-Xmx1G",
-#     "-jar",
-#     "/path/to/eclipse.jdt.ls/repository/plugins/org.eclipse.equinox.launcher_1.6.100.v20201223-0822.jar",
-#     "-configuration",
-#     "/path/to/eclipse.jdt.ls/repository/config_linux",
-#     "-data",
-#     "/path/to/eclipse-workspace",
-# ]
-# [language.java.settings]
-# # See https://github.dev/eclipse/eclipse.jdt.ls
-# # "java.format.insertSpaces" = true
-
-[language.jsx] # works for javascript as well
```

Comment:
The reason why this uses `jsx` instead of `javascript` is a bit hidden,
because it was documented only for [`tsx` instead of `typescript`](https://github.com/kak-lsp/kak-lsp/commit/e8f109d7da52aa9a6c4586dcbc6d4bae2ae47c77).
Essentially `jsx` (perhaps that's the same as `javascriptreact`?) is a
superset of javascript, so it works for plain javascript.
On the other hand, `language_id=javascript` doesn't work for *.jsx files.
So I think what we should do here is add these mappings

	[language_ids]
	javascript = "jsx"
	typescript = "tsx"

unless there is a reason for changing the status quo.. I don't know much about frontend dev.
(I'm not sure if `javascriptreact` is better/worse)

Reply:
ok looks like you're right; I've read elsewhere that `javascriptreact` is the proper language ID and not `jsx`